src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -295,11 +262,24 @@ public void clear(String feedId) {
   }
 
   /**
-   * Removes the latest added trip pattern from the cache. This should be done when removing the
-   * trip times from the timetable the trip has been added to.
+   * Removes previous trip-update from buffer if there is an update with given trip on service date
+   *
+   * @param serviceDate service date
```

Comment:
It feels to me like this Javadoc and the name of this method might be misrepresenting its purpose. Please confirm if my understanding is correct. The method doesn't exactly remove a trip update, it removes some objects created only if a trip update with very particular characteristics has been applied. These are specifically a realtime-created TripPattern and the single trip that is expected to be present on that same TripPattern. And crucially, it only performs this removal if the supplied tripId has previously caused a new TripPattern to be created (conditional on the `pattern != null` block), which only happens when the stop sequence is changed by the update.

The Javadoc (and maybe the name) gives the impression that this method takes an action if any update has ever been applied to the given trip and service date combination, but the actual field of application is much narrower.

Aside: the `@param serviceDate service date` annotation seems trivial/redundant and should probably be removed.

Reply:
Yes I agree with. I copied over the javadoc when I moved this method from the `SiriTimetableSnapshotSource` to here. I can update the javadoc to better match what the method does.

Reply:
> Aside: the @param serviceDate service date annotation seems trivial/redundant and should probably be removed. 
+1